### PR TITLE
[IE/DML]Implement InstanceNorm

### PR DESCRIPTION
**Pull request by lisa0314**
- Linux
out/ie/webnn_end2end_tests --gtest_filter=InstanceNormTests.*
Note: Google Test filter = InstanceNormTests.*
[==========] Running 3 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 3 tests from InstanceNormTests
[ RUN      ] InstanceNormTests.InstanceNormNchw
[       OK ] InstanceNormTests.InstanceNormNchw (42 ms)
[ RUN      ] InstanceNormTests.InstanceNormNhwc
[       OK ] InstanceNormTests.InstanceNormNhwc (35 ms)
[ RUN      ] InstanceNormTests.InstanceNormWithEpsilon
[       OK ] InstanceNormTests.InstanceNormWithEpsilon (7 ms)
[----------] 3 tests from InstanceNormTests (84 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test suite ran. (84 ms total)
[  PASSED  ] 3 tests.

- Windows
out\dml\webnn_end2end_tests.exe
--gtest_filter=InstanceNormTests.*
Note: Google Test filter = InstanceNormTests.*
[==========] Running 3 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 3 tests from InstanceNormTests
[ RUN ] InstanceNormTests.InstanceNormNchw
[ OK ] InstanceNormTests.InstanceNormNchw (130 ms)
[ RUN ] InstanceNormTests.InstanceNormNhwc
[ OK ] InstanceNormTests.InstanceNormNhwc (125 ms)
[ RUN ] InstanceNormTests.InstanceNormWithEpsilon
[ OK ] InstanceNormTests.InstanceNormWithEpsilon (31 ms)
[----------] 3 tests from InstanceNormTests (286 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test suite ran. (286 ms total)
[ PASSED ] 3 tests.

### CODE REVIEW

File: webnn.json [hunk 1]
```diff
@@ -433,6 +442,15 @@
...
       },
+      {
+        "name": "instance norm",
+        "returns": "operand",
+        "args": [
+          {"name": "input", "type": "operand"},
+          {"name": "options", "type": "instanceNorm options", "annotation": "const*", "optional": true}
+        ]
+      },
+
```
Comment (huningxin) [inline webnn.json:453]: please remove the blank line.

**Review by huningxin — APPROVED** [review]
LGTM with a nit. @lisa0314 , please help fix it and we can merge. Thanks!

**Comment by lisa0314** [comment]
@huningxin Code update. Please help to merge it, thanks!

File: src/webnn_native/dml/GraphDML.cpp [hunk 2]
```diff
@@ -1082,6 +1082,71 @@ namespace webnn_native { namespace dml {
...
+            // Set 1 to automatically broadcast those dimensions across the input.
+            ::dml::TensorDimensions expandDimens(4, 1);
+            expandDimens[1] = dimensions[0];
+            auto expandStrides = CalculateStrides(expandDimens);
+            expressions.push_back(::dml::Reinterpret(expression, expandDimens, expandStrides));
+        }
+        // Set tensor's dimensions to {1, channel, 1, 1} if scale or bias is null.
+        const DML_TENSOR_DATA_TYPE type = DML_TENSOR_DATA_TYPE_FLOAT32;
+        if (options->scale == nullptr) {
+            float* scale = (float*)malloc(sizeof(float) * inputDims[1]);
```
Comment (fujunwei) [inline src/webnn_native/dml/GraphDML.cpp:1121]: Maybe you can try to use `std::vector<float> scale(inputDims[1], 1.0) `instead L.1121 - L 1124.

File: src/webnn_native/dml/GraphDML.cpp [hunk 3]
```diff
@@ -1082,6 +1082,71 @@ namespace webnn_native { namespace dml {
...
+            for (size_t i = 0; i < inputDims[1]; i++) {
+                scale[i] = 1;
+            }
+            expressions.insert(expressions.begin(),
+                               BindingConstant(type, {1, inputDims[1], 1, 1}, scale,
+                                               sizeof(float) * inputDims[1]));
+            free(scale);
+        }
+        if (options->bias == nullptr) {
+            float* bias = (float*)malloc(sizeof(float) * inputDims[1]);
```
Comment (fujunwei) [inline src/webnn_native/dml/GraphDML.cpp:1131]: Ditto.

File: src/webnn_native/ops/InstanceNorm.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        if (options != nullptr) {
+            mOptions = *options;
+            if (options->scale != nullptr) {
+                mInputs.push_back(options->scale);
+            }
+            if (options->bias != nullptr) {
+                mInputs.push_back(options->bias);
+            }
+        } else {
+            mOptions.scale = nullptr;
```
Comment (fujunwei) [inline src/webnn_native/ops/InstanceNorm.cpp:36]: How abouting adding the default nullptr in `webnn.json`？

File: src/webnn_native/dml/GraphDML.cpp (hunk 2 above)
Comment (lisa0314) [reply src/webnn_native/dml/GraphDML.cpp:1121]: Got it.

**Comment by huningxin** [comment]
@miaobin , are you able to take a look at the IC failure? Thanks.

**Comment by huningxin** [comment]
Transferred to https://github.com/webmachinelearning/webnn-native/pull/39, thanks @miaobin !

**Pull request closed without merging**